Validate documentation

JPEWdev opened the pull request:
Validates security documentation

goneall reviewed (approved):
Changes LGTM and it does pass the CI with the additional checking.

maxhbr commented on docs/annexes/cross-reference.md:
````diff
@@ -0,0 +1,209 @@
+Markdown file, use the following command:
+
+```shell
+cat getting-started.md | awk 'BEGIN{flag=0} /^```json/, $0=="```" { if (/^---$/){flag++} else if ($0 !~ /^```.*/ ) print $0 > "doc-" flag ".spdx.json"}'
````
```suggestion
cat cross-reference.md | awk 'BEGIN{flag=0} /^```json/, $0=="```" { if (/^---$/){flag++} else if ($0 !~ /^```.*/ ) print $0 > "doc-" flag ".spdx.json"}'
```

maxhbr commented:
I just added the functionality to load multiple documents to my tool https://github.com/maxhbr/spdx3ToGraph and this generates the following for the cross-reference.md example:

![doc-0 spdx json](https://github.com/spdx/spdx-spec/assets/1187050/466f0cd3-ac9c-421f-9013-eee1adf160eb)

JPEWdev commented:
> I just added the functionality to load multiple documents to my tool https://github.com/maxhbr/spdx3ToGraph and this generates the following for the cross-reference.md example:
> 
> ![doc-0 spdx json](https://private-user-images.githubusercontent.com/1187050/329196831-466f0cd3-ac9c-421f-9013-eee1adf160eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNjM2MTMsIm5iZiI6MTcxNTI2MzMxMywicGF0aCI6Ii8xMTg3MDUwLzMyOTE5NjgzMS00NjZmMGNkMy1hYzljLTQyMWYtOTAxMy1lZWUxYWRmMTYwZWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDlUMTQwMTUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGNjMmM1MDVhNjg3YjI1ODgwNTI5NTk4ODJhM2E1YjBjMjQwNmRjMGFmNmZiYTgxNzVhNTMyMzhlMTRjYjRjYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1cvm3eMWg4wLo-8HGHfu55yAYI8KDpw4UEo0z1s54c8)

Very cool!

JPEWdev commented:
> I just added the functionality to load multiple documents to my tool https://github.com/maxhbr/spdx3ToGraph and this generates the following for the cross-reference.md example:
> 
> ![doc-0 spdx json](https://private-user-images.githubusercontent.com/1187050/329196831-466f0cd3-ac9c-421f-9013-eee1adf160eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNjM2MTMsIm5iZiI6MTcxNTI2MzMxMywicGF0aCI6Ii8xMTg3MDUwLzMyOTE5NjgzMS00NjZmMGNkMy1hYzljLTQyMWYtOTAxMy1lZWUxYWRmMTYwZWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDlUMTQwMTUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGNjMmM1MDVhNjg3YjI1ODgwNTI5NTk4ODJhM2E1YjBjMjQwNmRjMGFmNmZiYTgxNzVhNTMyMzhlMTRjYjRjYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1cvm3eMWg4wLo-8HGHfu55yAYI8KDpw4UEo0z1s54c8)

This made me realize I forgot the `serialziedInArtifact` relationship, so I added that

maxhbr commented:
update:

![doc-0 spdx json](https://github.com/spdx/spdx-spec/assets/1187050/63825a51-0dca-430a-9c6c-38bb6373b36b)

maxhbr commented on docs/annexes/cross-reference.md:
```diff
@@ -0,0 +1,246 @@
+        },
+        {
+            "type": "SpdxDocument",
+            "spdxId": "http://spdx.example.com/Document1",
```
I think we should use more realistic and potentially globally unique IDs.

JPEWdev replied:
What classifies as "more realistic"?

maxhbr replied:
something that could be globally unique. Maybe what I am missing is guidance on how to create IDs. I just know hand crafted non-realistic examples. Some using `urn:..`. I think this might be something for a different document but I it is a gap.

JPEWdev replied:
Ya, I don't mind making it better, I'm just not clear what "better" would be in this case. There's not currently great guidance on this other than that it should be "unique"

goneall reviewed (approved):
Very nice explanation - Thanks @JPEWdev

kestewart reviewed (approved):
Thanks for adding this in Joshua!
